### PR TITLE
Upvote answer (closes #54)

### LINKED ISSUE #54: API: Upvote an Answer
As a developer I want to have an option to add a vote for an answer or remove a previous one.

### PR DESCRIPTION
**Pull request by zsoltime**
I haven't added tests yet as I'm waiting for my other PR (#68) to be accepted first.

### CODE REVIEW

**Review by Heyjp — COMMENTED** [review]
Commented on the wrong PR 😂 

At the moment we are returning an array of user_id's is it possible that we instead have mongoose just return a count of the length of the array in place of that array? I think that voting should be anonymous at least to other users.

**Comment by zsoltime** [comment]
And I was wondering why I can't send my response 😃 

Anyway, I need to check if we can display the length of the array, in that case we don't even need to have a `votes` field with the number of votes. We only need to know two things on the front end; if the user has voted already and the number of votes. I'll look into this, but any help is appreciated. 😄

**Comment by Heyjp** [comment]
So this will return an object with just the counted votes. You can add and remove fields by just listing that field as either `0` or `1`.

```
exports.vote = async (ctx) => {

 await Answer.aggregate([{ $match: { _id: ObjectId(ctx.params.id) } },
    {
      $project: {
        _id: 0,
        votes: { $size: '$voted_by' },
      },
    },
  ]).then((res) => {
    ctx.body = res[0];
    return res;
  });
}
```

**Comment by zsoltime** [comment]
So you can use the mongo aggregation framework with Mongoose? That's great, thanks a lot 😃

**Comment by Heyjp** [comment]
Branch has conflicts before it can be merged.

**Comment by zsoltime** [comment]
@Heyjp Fixed, and added tests

File: server/controller/answers.js
Comment (Heyjp) [inline server/controller/answers.js]: I don't think async needs to be here. You could simply return `Answer.findOneAndUpdate` and do another `.then()` call as the function is already waiting for `Answer.findOne` to resolve.

**Review by Heyjp — CHANGES_REQUESTED** [review]
Just the one comment about the use of async. The rest works well.

**Comment by Heyjp** [comment]
Going to merge this anyway now and review it later.

**Review by Heyjp — APPROVED** [review]

**Pull request merged**
